### DISCUSSION TITLE
Select demosaicing algorithm

### DISCUSSION THREAD

**Discussion by HectorFP** [dlemstra/Magick.NET#1060] [Help] [answered]
Hi,

I'm developing a small app to test the performance (memory, CPU and time) and the differences between demosaicing algorithms in .NET (+Gtk), and I decided to use Magick.NET, because it seemed like a very good option and the best in quantity of supported formats.

I have implemented everything already, but I don't know how to set/change the demosaicing algorithm to use for RAW file conversions.

The Libraw documentation indicates that the user_qual field is the same as -q working with dcraw, but I have not found anything similar in the Magick.NET source code.

int user_qual;
dcraw keys: -q
0-10: interpolation quality:
0 - linear interpolation
1 - VNG interpolation
2 - PPG interpolation
3 - AHD interpolation
4 - DCB interpolation
11 - DHT intepolation
12 - Modified AHD intepolation (by Anton Petrusevich)

Is there a way to select the algorithm to use?

Thanks!

**Comment by dlemstra** [comment, marked as answer]
We don't have this option available yet. I just pushed a change to make this possible in ImageMagick. And I will add an option to make this available in the next release of Magick.NET by a new property called `InterpolationQuality` in the `DngReadDefines` class.

**Reply by HectorFP** [reply]
I'll wait for that property :)

But could it be possible to set "dng: interpolation-quality" via SetDefine?

Thanks for your support

**Reply by dlemstra** [reply]
You can also set it through a define if you prefer that. But that define was just added and will become available in version `8.4.0` of MagickNET.

**Reply by dlemstra** [reply]
The new release has just been published.

**Reply by HectorFP** [reply]
Thanks!

And another question related to interpolation of colors, is it possible to apply / use nearest neighbor (NN)? because I have taken a look at the algorithms supported by Libraw and it seems that it is not an option

**Reply by dlemstra** [reply]
We can only support what is available by Libraw when reading a `DNG` file.
